# import statements

In [1]:
import pandas as pd
import numpy as np 
from collections import Counter

# code to read student.csv into pandas dataframe 

In [12]:
df = pd.read_csv('student.csv', delimiter = ';')


# helper functions (if needed)

In [148]:

print(df.columns)
print(df['G3'])

Index(['school', 'sex', 'age', 'address', 'famsize', 'Pstatus', 'Medu', 'Fedu',
       'Mjob', 'Fjob', 'reason', 'guardian', 'traveltime', 'studytime',
       'failures', 'schoolsup', 'famsup', 'paid', 'activities', 'nursery',
       'higher', 'internet', 'romantic', 'famrel', 'freetime', 'goout', 'Dalc',
       'Walc', 'health', 'absences', 'G1', 'G2', 'G3'],
      dtype='object')
0       6
1       6
2      10
3      15
4      10
       ..
390     9
391    16
392     7
393    10
394     9
Name: G3, Length: 395, dtype: int64


# Python code to answer question 1) What is the minimum, maximum, average, and median final grade (G3 column)?

In [149]:
q1 = df['G3']


print("G3 column minimum final grade ")
min= np.min(q1)
print(min)

print("G3 column maximum final grade ")
max= np.max(q1)
print(max)

print("G3 column median final grade ")
medi= np.median(q1)
print(medi)

print("G3 column average final grade ")
avg = np.mean(q1)
print(avg)


G3 column minimum final grade 
0
G3 column maximum final grade 
20
G3 column median final grade 
11.0
G3 column average final grade 
10.415189873417722


# Text answer for question 1.

G3 column minimum final grade 0
G3 column maximum final grade 20
G3 column median final grade 11.0
G3 column average final grade 10.415189873417722

# Python code to answer question 2) Which subgroup of sex, male or female, had the highest average final grade (G3)?

In [150]:
df.groupby(['sex'])['G3'].mean()

sex
F     9.966346
M    10.914439
Name: G3, dtype: float64

# Text answer for question 2.  

F     9.966346
M    10.914439

# Python code to answer question 3) Which subgroup of Fjob had the lowest average final grade?

In [151]:
subgroup = df.groupby(['Fjob'])['G3'].mean()
print(subgroup.head(1))

Fjob
at_home    10.15
Name: G3, dtype: float64


# Text answer for question 3.

at_home    10.15

# Python code to answer question 4) On average, did students in romantic relationships get higher or lower final grades?

In [152]:
subgroup = df.groupby(['romantic'])['G3'].mean()
print(subgroup)

romantic
no     10.836502
yes     9.575758
Name: G3, dtype: float64


# Text answer for question 4.

no     10.836502
yes     9.575758
on average no romantic people got a better final grade

# Python code to answer question 5) On average, do people with internet access at home make higher or lower final grades than people without internet? 

In [153]:
subgroup = df.groupby(['internet'])['G3'].mean()



# Text answer for question 5.

no      9.409091
yes    10.617021
on average those with internet at home got a better final grade

# Python code to answer question 6) What percentage of male students are in romantic relationships?  What percentage of female students are in romantic relationships?

In [154]:
romantic = df['romantic'] == 'yes'
x = df['sex'][romantic]
romantic = x == 'M'
counts = Counter(romantic)
num = counts[True]
perc = 100 * ( num / len(df))
print("male")
print(perc)


romantic = df['romantic'] == 'yes'
x = df['sex'][romantic]
romantic = x == 'F'
counts = Counter(romantic)
num = counts[True]
perc = 100 * ( num / len(df))
print("female")
print(perc)


male
13.41772151898734
female
20.0


# Text answer for question 6.

male
47.34177215189874

female
52.65822784810127

# Python code to answer question 7) What is the correlation between freetime and final grade? 

In [155]:
corr = df['freetime'].corr(df['G3'])
print(corr)

0.011307240308313022


# Text answer for question 7.

 correlation between freetime and final grade is 0.011307240308313022

# Python code to answer question 8) What is the correlation between studytime and final grade? 

In [156]:
corr = df['studytime'].corr(df['G3'])
print(corr)

0.09781968965319633


# Text answer for question 8.

the correlation between studytime and final grade is 0.09781968965319633

# Python code to answer question 9) What is the correlation between studytime and freetime? 

In [157]:
corr = df['studytime'].corr(df['freetime'])
print(corr)

-0.14319840716795168


# Text answer for question 9.

the correlation between studytime and freetime is -0.14319840716795168

# Python code to answer question 10) Excluding columns G1 and G2, which numerical column had the biggest correlation with final grade (G3)?

In [158]:

#for (columnName, columnData) in df.iteritems():
#        name = "'" + columnName + "'"
#        name2 = df[name]
#        if name == 'G1':
#            break
#        else:
 #           print(name)
 #           corr = df[name2].corr(df['G3'])

print("school")
corr = df['school'].corr(df['G3'])
print(corr)
print("sex")
corr = df['sex'].corr(df['G3'])
print(corr)
print("age")
corr = df['age'].corr(df['G3'])
print(corr)
print("address")
corr = df['address'].corr(df['G3'])
print(corr)
print("famsize")
corr = df['famsize'].corr(df['G3'])
print(corr)
print("Pstatus")
corr = df['Pstatus'].corr(df['G3'])
print(corr)
print("Medu")
corr = df['Medu'].corr(df['G3'])
print(corr)
print("Fedu")
corr = df['Fedu'].corr(df['G3'])
print(corr)
print("Mjob")
corr = df['Mjob'].corr(df['G3'])
print(corr)
print("Fjob")
corr = df['Fjob'].corr(df['G3'])
print(corr)
print("reason")
corr = df['reason'].corr(df['G3'])
print(corr)
print("guardian")
corr = df['guardian'].corr(df['G3'])
print(corr)
print("traveltime")
corr = df['traveltime'].corr(df['G3'])
print(corr)
print("studytime")
corr = df['studytime'].corr(df['G3'])
print(corr)
print("failures")
corr = df['failures'].corr(df['G3'])
print(corr)
print("schoolsup")
corr = df['schoolsup'].corr(df['G3'])
print(corr)
print("famsup")
corr = df['famsup'].corr(df['G3'])
print(corr)
print("paid")
corr = df['paid'].corr(df['G3'])
print(corr)
print("activities")
corr = df['activities'].corr(df['G3'])
print(corr)
print("nursery")
corr = df['nursery'].corr(df['G3'])
print(corr)
print("higher")
corr = df['higher'].corr(df['G3'])
print(corr)
print("internet")
corr = df['internet'].corr(df['G3'])
print(corr)
print("romantic")
corr = df['romantic'].corr(df['G3'])
print(corr)
print("famrel")
corr = df['famrel'].corr(df['G3'])
print(corr)
print("freetime")
corr = df['freetime'].corr(df['G3'])
print(corr)
print("goout")
corr = df['goout'].corr(df['G3'])
print(corr)
print("prin")
corr = df['prin'].corr(df['G3'])
print(corr)
print("Walc")
corr = df['Walc'].corr(df['G3'])
print(corr)
print("health")
corr = df['health'].corr(df['G3'])
print(corr)
print("absences")
corr = df['absences'].corr(df['G3'])
print(corr)


school


TypeError: unsupported operand type(s) for /: 'str' and 'int'

# Text answer for question 10.  Please identify the column and correlation.

text...